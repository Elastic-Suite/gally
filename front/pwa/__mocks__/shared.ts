import { api } from 'shared/src/mocks'

import categorySortingOptions from 'shared/src/mocks/static/category_sorting_options.json'
import docs from 'shared/src/mocks/static/docs.json'
import entrypoint from 'shared/src/mocks/static/index.json'
import graphql from 'shared/src/mocks/static/graphql.json'
import metadata from 'shared/src/mocks/static/metadata.json'
import sourceFieldOptionLabels from 'shared/src/mocks/static/source_field_option_labels.json'

export * from 'shared/src/constants'
export * from 'shared/src/contexts'
export * from 'shared/src/mocks'
export * from 'shared/src/services/breadcrumb'
export * from 'shared/src/services/catalog'
export * from 'shared/src/services/category'
export * from 'shared/src/services/field'
export * from 'shared/src/services/format'
export * from 'shared/src/services/hydra'
export * from 'shared/src/services/local'
export * from 'shared/src/services/options'
export * from 'shared/src/services/rules'
export * from 'shared/src/services/style'
export * from 'shared/src/services/table'
export * from 'shared/src/services/url'
export * from 'shared/src/services/user'
export * from 'shared/src/types'

const body = { hello: 'world' }

/* api */
export const getApiUrl = jest.fn((url) => url)

export const fetchApi = jest.fn((_, resource) => {
  let data: unknown = { ...body }
  if (
    (typeof resource !== 'string' &&
      resource.title.toLowerCase() === 'metadata') ||
    (typeof resource === 'string' && resource.endsWith('metadata'))
  ) {
    data = { ...metadata }
  } else if (
    typeof resource === 'string' &&
    resource.endsWith('category_sorting_options')
  ) {
    data = { ...categorySortingOptions }
  } else if (
    (typeof resource !== 'string' &&
      resource.title.toLowerCase() === 'source_field_option_label') ||
    (typeof resource === 'string' &&
      resource.endsWith('source_field_option_labels'))
  ) {
    data = { ...sourceFieldOptionLabels }
  }
  return Promise.resolve(data)
})

export const removeEmptyParameters = jest.fn(
  (searchParameters = {}) => searchParameters
)

export const log = jest.fn((error, log) => log(error.message))

export const getApiFilters = jest.fn((x) => x)

/* fetch */
export const isError = jest.fn((json) => 'error' in json)
export const fetchJson = jest.fn((url) => {
  switch (url) {
    case 'http://localhost/':
      return Promise.resolve({
        json: entrypoint,
        response: {
          headers: new Headers({
            Link: '<http://localhost/docs.jsonld>; rel="http://www.w3.org/ns/hydra/core#apiDocumentation"',
          }),
        },
      })
    case 'http://localhost/docs.jsonld':
      return Promise.resolve({
        json: docs,
      })
    case 'http://localhost/graphql':
      return Promise.resolve({
        json: graphql,
      })
    default:
      return Promise.resolve({
        json: body,
        response: { headers: new Headers() },
      })
  }
})

/* form */

export const getFormValidityError = jest.fn(() => 'test')

/* graphql */
export const fetchGraphql = jest.fn(() => {
  const data: unknown = { ...body }
  return Promise.resolve(data)
})

/* parser */
export const parseSchema = jest.fn(() => Promise.resolve(api))

/* storage */
export const storageGet = jest.fn()
export const storageSet = jest.fn()
export const storageRemove = jest.fn()

/* useSchemaLoader */
export const useSchemaLoader = jest.fn(() => ({ data: 'api' }))
