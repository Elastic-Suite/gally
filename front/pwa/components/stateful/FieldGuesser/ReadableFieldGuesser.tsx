import { Switch } from '@mui/material'
import { styled } from '@mui/system'
import { useTranslation } from 'next-i18next'
import {
  DataContentType,
  IFieldGuesserProps,
  IPrice,
  IScore,
  IStock,
} from 'gally-admin-shared'

import Chip from '~/components/atoms/Chip/Chip'
import Score from '~/components/atoms/score/Score'
import Stock from '~/components/atoms/stock/Stock'
import Price from '~/components/atoms/price/Price'

const Image = styled('img')({
  height: 80,
  width: 80,
})

const Box = styled('div')({
  height: 40,
  display: 'flex',
  alignItems: 'center',
  overflow: 'hidden',
  textOverflow: 'ellipsis',
  whiteSpace: 'nowrap',
})

function ReadableFieldGuesser(props: IFieldGuesserProps): JSX.Element {
  const { input, value } = props
  const { t } = useTranslation('common')

  if (value === undefined || value === null) {
    return null
  }

  switch (input) {
    case DataContentType.BOOLEAN: {
      return <Switch disabled defaultChecked={value as boolean} />
    }

    case DataContentType.TAG: {
      return <Chip label={value as string} />
    }

    case DataContentType.IMAGE: {
      return <Image alt={t('field.productImage')} src={value as string} />
    }

    case DataContentType.SCORE: {
      const score = { scoreValue: value } as IScore
      return <Score scoreValue={score.scoreValue} />
    }

    case DataContentType.STOCK: {
      return <Stock stockStatus={(value as IStock).status} />
    }

    case DataContentType.PRICE: {
      if (!value) {
        return null
      }
      const [{ price }] = value as IPrice[]
      // todo : how backend will handle currency ?
      return <Price price={price} countryCode="fr-FR" currency="EUR" />
    }

    default: {
      return <Box>{value as string}</Box>
    }
  }
}

export default ReadableFieldGuesser
