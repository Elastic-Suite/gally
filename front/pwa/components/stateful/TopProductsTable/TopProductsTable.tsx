import { styled } from '@mui/system'
import { useTranslation } from 'next-i18next'
import TopProductsBanner from '~/components/molecules/CustomTable/TopProductsBanner/TopProductsBanner'
import CustomTable, {
  ICustomTableProps,
} from '../../organisms/CustomTable/CustomTable'

const Container = styled('div')(({ theme }) => ({
  borderRadius: theme.shape.borderRadius,
  overflow: 'hidden',
}))

function TopProductsTable(props: ICustomTableProps): JSX.Element {
  const { t } = useTranslation('common')

  return (
    <Container>
      <TopProductsBanner bannerText={t('searchResults.topProducts')} />
      <CustomTable {...props} />
    </Container>
  )
}

export default TopProductsTable
