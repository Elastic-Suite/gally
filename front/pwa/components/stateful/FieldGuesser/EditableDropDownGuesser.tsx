import { SyntheticEvent, useContext, useEffect } from 'react'
import { useTranslation } from 'next-i18next'
import { IFieldGuesserProps, LoadStatus } from 'shared'

import DropDown from '~/components/atoms/form/DropDown'
import { optionsContext } from '~/contexts'

interface IProps extends Omit<IFieldGuesserProps, 'onChange'> {
  onChange: (
    value: number | string | (number | string)[],
    event: SyntheticEvent
  ) => void
}

function EditableDropDownGuesser(props: IProps): JSX.Element {
  const {
    diffValue,
    field,
    label,
    multiple,
    onChange,
    options,
    required,
    value,
  } = props

  const { t } = useTranslation('common')
  const { fieldOptions, load, statuses } = useContext(optionsContext)
  const dropDownOptions =
    options ?? fieldOptions.get(field.property['@id']) ?? []
  const dirty = diffValue !== undefined && diffValue !== value

  useEffect(() => {
    if (!options && field) {
      load(field)
    }
  }, [field, load, options])

  // Wait to load the options before rendering to avoid problems
  if (
    value &&
    dropDownOptions.length === 0 &&
    statuses.current.get(field.property['@id']) !== LoadStatus.SUCCEEDED
  ) {
    return null
  }

  return (
    <DropDown
      dirty={dirty}
      helperText={
        Boolean(dirty) &&
        t('form.defaultValue', {
          value: dropDownOptions.find(({ value }) => value === diffValue)
            ?.label,
        })
      }
      label={label}
      multiple={multiple}
      options={dropDownOptions}
      required={required}
      value={value}
      onChange={onChange}
    />
  )
}

export default EditableDropDownGuesser
