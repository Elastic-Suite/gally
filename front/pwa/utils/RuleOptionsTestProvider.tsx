import { ReactNode } from 'react'

import ruleEngineOperator from '~/public/mocks/rule_engine_operators.json'
import sourceFields from '~/public/mocks/source_fields.json'
import sourceFieldLabels from '~/public/mocks/source_field_labels.json'
import { RuleAttributeType } from 'shared'

import RuleOptionsProvider from '~/components/stateful-providers/RuleOptionsProvider/RuleOptionsProvider'

interface IProps {
  children: ReactNode
}

const sourceFieldLabelsMap = new Map(
  sourceFieldLabels['hydra:member'].map((label) => [
    label.sourceField,
    label.label,
  ])
)

const fields = sourceFields['hydra:member'].map((field) => {
  const label = sourceFieldLabelsMap.get(`/source_fields/${field.id}`)
  return {
    id: field.id,
    code: field.code,
    label: (label || field.defaultLabel) ?? '',
    type: field.type as RuleAttributeType,
  }
})

function RuleOptionsTestProvider(props: IProps): JSX.Element {
  const { children } = props
  return (
    <RuleOptionsProvider
      catalogId={-1}
      localizedCatalogId={-1}
      fields={fields}
      ruleOperators={ruleEngineOperator}
    >
      {children}
    </RuleOptionsProvider>
  )
}

export default RuleOptionsTestProvider
