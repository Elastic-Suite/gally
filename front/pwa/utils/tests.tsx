import { RenderOptions, render, renderHook } from '@testing-library/react'
import { PreloadedState } from '@reduxjs/toolkit'
import { ReactNode } from 'react'
import { Bundle, IUser, api } from 'shared'

import { AppStore, RootState, setupStore } from '~/store'

import AppProvider from '~/components/stateful-providers/AppProvider/AppProvider'

import TestProvider from './TestProvider'

export interface IExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: PreloadedState<RootState>
  store?: AppStore
}

export interface IRenderOutput {
  store: AppStore
}

const timestamp = Math.floor(Date.now() / 1000)
const expires = 8 * 60 * 60 // 8 hours

const user = {
  iat: timestamp,
  exp: timestamp + expires,
  roles: ['ROLE_ADMIN', 'ROLE_CONTRIBUTOR'],
  username: 'admin@example.com',
} as IUser

const defaultState = {
  data: {
    api,
    bundles: [Bundle.VIRTUAL_CATEGORY],
    configurations: ['https://localhost/products'],
  },
  user: {
    requestedPath: '',
    token:
      'eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiJ9.eyJpYXQiOjE2NTk5NTM1NDQsImV4cCI6MTY1OTk4MjM0NCwicm9sZXMiOlsiUk9MRV9BRE1JTiIsIlJPTEVfQ09OVFJJQlVUT1IiXSwidXNlcm5hbWUiOiJhZG1pbkBleGFtcGxlLmNvbSJ9.ZNE2SHmtm9WgWEMmKGMjpcfEym2ysxC_CVcLtzlpuT3jlT0Zyz4VjoQHdTav9I7s9b2QzpeDNt3Sfrbka8IHcVb_TvZle_vII-TkQGal37RtXxKk_EMMQQlzLRO3zDWYy2I0ZPXv6iMlaSWOya4egOG2s1_tzFf4MnNNm2zEQbjzAtZ_Ij9UBMlSxK19AHvHlKU9AelA-NepjUdl6f4GxYPnyIwZw3PS294Wtffj7Kzw1ZZywDljl1xN_flg70cuO0GGkUqBNHrf7zBjPmXbAZZDAefoyQfVqB2v4GzDJfO847XymvBlHtCCjWx0ThK5afyOGL2MAAZkl-AoNjMcjA',
    user,
  },
}

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/explicit-function-return-type
export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = defaultState,
    store,
    ...renderOptions
  }: IExtendedRenderOptions = {}
) {
  store = store ?? setupStore(preloadedState)
  function Wrapper({ children }: { children: ReactNode }): JSX.Element {
    return (
      <AppProvider store={store}>
        <TestProvider>{children}</TestProvider>
      </AppProvider>
    )
  }
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) }
}

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/explicit-function-return-type
export function renderHookWithProviders<R>(
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  hook: (props: any) => R,
  {
    preloadedState = defaultState,
    store,
    ...renderOptions
  }: IExtendedRenderOptions = {}
) {
  store = store ?? setupStore(preloadedState)
  function Wrapper({ children }: { children: ReactNode }): JSX.Element {
    return (
      <AppProvider store={store}>
        <TestProvider>{children}</TestProvider>
      </AppProvider>
    )
  }
  return { store, ...renderHook(hook, { wrapper: Wrapper, ...renderOptions }) }
}
