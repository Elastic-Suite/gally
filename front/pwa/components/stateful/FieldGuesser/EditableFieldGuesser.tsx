import { ChangeEvent, SyntheticEvent } from 'react'
import { useTranslation } from 'next-i18next'
import { DataContentType, IFieldGuesserProps } from 'shared'

import DropDown from '~/components/atoms/form/DropDown'
import InputTextError from '~/components/atoms/form/InputTextError'
import RangeError from '~/components/atoms/form/RangeError'
import Switch from '~/components/atoms/form/Switch'

import ReadableFieldGuesser from './ReadableFieldGuesser'
import EditableDropDownGuesser from './EditableDropDownGuesser'

function EditableFieldGuesser(props: IFieldGuesserProps): JSX.Element {
  const {
    diffValue,
    input,
    disabled,
    label,
    multiple,
    name,
    onChange,
    options,
    useDropdownBoolean,
    value,
    required,
    showError,
    suffix,
    type,
    validation,
  } = props
  const { t } = useTranslation('common')
  const dirty = diffValue !== undefined && diffValue !== value

  function handleChange(
    value: number | string | (number | string)[],
    event: SyntheticEvent
  ): void {
    if (onChange) {
      onChange(name, value, event)
    }
  }

  function handleSwitchChange(event: ChangeEvent<HTMLInputElement>): void {
    if (onChange) {
      onChange(name, event.target.checked, event)
    }
  }

  switch (input ?? type) {
    case DataContentType.NUMBER:
    case DataContentType.STRING: {
      return (
        <InputTextError
          dirty={dirty}
          disabled={disabled}
          helperText={
            Boolean(dirty) && t('form.defaultValue', { value: diffValue })
          }
          inputProps={validation}
          label={label}
          onChange={handleChange}
          required={required}
          showError={showError}
          suffix={suffix}
          type={input === DataContentType.NUMBER ? 'number' : 'text'}
          value={String(value)}
        />
      )
    }

    case DataContentType.RANGE: {
      return (
        <RangeError
          dirty={dirty}
          disabled={disabled}
          helperText={
            Boolean(dirty) && t('form.defaultValue', { value: diffValue })
          }
          inputProps={validation}
          label={label}
          onChange={handleChange}
          required={required}
          showError={showError}
          suffix={suffix}
          value={value as (string | number)[]}
        />
      )
    }

    case DataContentType.SELECT: {
      return <EditableDropDownGuesser {...props} onChange={handleChange} />
    }

    case DataContentType.BOOLEAN: {
      if (useDropdownBoolean) {
        return (
          <DropDown
            dirty={dirty}
            disabled={disabled}
            helperText={
              Boolean(dirty) && t('form.defaultValue', { value: diffValue })
            }
            label={label}
            multiple={multiple}
            options={
              options ?? [
                { label: t('filter.yes'), value: true },
                { label: t('filter.no'), value: false },
              ]
            }
            required={required}
            value={value}
            onChange={handleChange}
          />
        )
      }
      return (
        <Switch
          checked={Boolean(value)}
          disabled={disabled}
          helperText={
            Boolean(dirty) &&
            t('form.defaultValue', { value: t(`switch.${diffValue}`) })
          }
          onChange={handleSwitchChange}
          required={required}
        />
      )
    }

    default: {
      return <ReadableFieldGuesser {...props} />
    }
  }
}

export default EditableFieldGuesser
