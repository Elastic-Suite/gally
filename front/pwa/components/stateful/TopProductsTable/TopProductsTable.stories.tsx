import { ComponentMeta, ComponentStory } from '@storybook/react'

import FieldGuesser from '../FieldGuesser/FieldGuesser'

import TopProductsTableComponent from './TopProductsTable'

export default {
  title: 'Organisms/TopProductsTable',
  component: TopProductsTableComponent,
} as ComponentMeta<typeof TopProductsTableComponent>

const Template: ComponentStory<typeof TopProductsTableComponent> = (args) => (
  <TopProductsTableComponent {...args} />
)

const tableHeaders = [
  {
    input: 'string',
    name: 'code',
    label: 'Code',
    type: 'string',
    editable: false,
    sticky: false,
  },
  {
    input: 'image',
    name: 'image',
    label: 'Image',
    type: 'image',
    editable: false,
    sticky: false,
  },
  {
    input: 'string',
    name: 'name',
    label: 'Name',
    type: 'string',
    editable: false,
    sticky: false,
  },
  {
    input: 'score',
    name: 'score',
    label: 'Score',
    type: 'score',
    editable: false,
    sticky: false,
  },
  {
    input: 'stock',
    name: 'stock',
    label: 'In stock',
    type: 'stock',
    editable: false,
    sticky: false,
  },
  {
    input: 'price',
    name: 'price',
    label: 'Price',
    type: 'price',
    editable: false,
    sticky: false,
  },
  {
    input: 'boolean',
    name: 'visible',
    label: 'Visible',
    type: 'boolean',
    editable: false,
    sticky: false,
  },
]
const tableRows = [
  {
    id: '1',
    code: 'VA03',
    image: 'static/media/assets/img/scarf_elastic.png',
    name: 'Product name',
    score: 100.11,
    stock: {
      status: true,
      qty: 10,
    },
    price: 10,
    visible: true,
  },
  {
    id: '2',
    code: 'VA03',
    image: 'static/media/assets/img/scarf_elastic.png',
    name: 'Product name',
    score: 100.11,
    stock: {
      status: false,
      qty: 10,
    },
    price: 10,
    visible: true,
  },
]

export const TopProductTable = Template.bind({})
TopProductTable.args = {
  Field: FieldGuesser,
  tableHeaders,
  tableRows,
}
