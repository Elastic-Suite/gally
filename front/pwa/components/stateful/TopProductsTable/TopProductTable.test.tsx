import { renderWithProviders } from '~/utils/tests'
import { DataContentType, ITableHeader, ITableRow } from 'shared'

import FieldGuesser from '../FieldGuesser/FieldGuesser'

import TopProductsTable from './TopProductsTable'

describe('TopProductTable', () => {
  it('Should match snapschot', () => {
    const tableHeaders: ITableHeader[] = [
      {
        input: DataContentType.STRING,
        name: 'code',
        label: 'Code',
        type: DataContentType.STRING,
        editable: false,
        sticky: false,
      },
      {
        input: DataContentType.IMAGE,
        name: 'image',
        label: 'Image',
        type: DataContentType.IMAGE,
        editable: false,
        sticky: false,
      },
      {
        input: DataContentType.STRING,
        name: 'name',
        label: 'Name',
        type: DataContentType.STRING,
        editable: false,
        sticky: false,
      },
      {
        input: DataContentType.SCORE,
        name: 'score',
        label: 'Score',
        type: DataContentType.SCORE,
        editable: false,
        sticky: false,
      },
      {
        input: DataContentType.STOCK,
        name: 'stock',
        label: 'In stock',
        type: DataContentType.STOCK,
        editable: false,
        sticky: false,
      },
      {
        input: DataContentType.PRICE,
        name: 'price',
        label: 'Price',
        type: DataContentType.PRICE,
        editable: false,
        sticky: false,
      },
      {
        input: DataContentType.BOOLEAN,
        name: 'visible',
        label: 'Visible',
        type: DataContentType.BOOLEAN,
        editable: false,
        sticky: false,
      },
    ]
    const tableRows: ITableRow[] = [
      {
        id: '1',
        code: 'VA03',
        image: 'static/media/assets/img/scarf_elastic.png',
        name: 'Product name',
        score: 100.11,
        stock: { status: true },
        price: [{ price: 10 }],
        visible: true,
      },
      {
        id: '2',
        code: 'VA03',
        image: 'static/media/assets/img/scarf_elastic.png',
        name: 'Product name',
        score: 100.11,
        stock: { status: false },
        price: [{ price: 10 }],
        visible: true,
      },
    ]
    const { container } = renderWithProviders(
      <TopProductsTable
        Field={FieldGuesser}
        tableHeaders={tableHeaders}
        tableRows={tableRows}
        selectedRows={[]}
        draggable
      />
    )
    expect(container).toMatchSnapshot()
  })
})
