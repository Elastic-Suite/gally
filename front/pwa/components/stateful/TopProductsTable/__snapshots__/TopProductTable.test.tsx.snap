// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopProductTable Should match snapschot 1`] = `
<div>
  <div
    class="css-uj88lt"
  >
    <div
      class="MuiBox-root css-1ndjkvn"
    >
      <ion-icon
        name="trending-up-outline"
      />
      <div
        class="MuiBox-root css-x43z6g"
      >
        searchResults.topProducts
      </div>
    </div>
    <div
      class="MuiTableContainer-root css-1i0gtxd-MuiTableContainer-root"
    >
      <table
        class="MuiTable-root css-1pty5za-MuiTable-root"
      >
        <thead
          class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-head css-1u1grnd-MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-8uqeip-MuiTableCell-root"
              scope="col"
            />
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1in94r8-MuiTableCell-root"
              scope="col"
            >
              Code
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1g6gaq7-MuiTableCell-root"
              scope="col"
            >
              Image
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1in94r8-MuiTableCell-root"
              scope="col"
            >
              Name
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-ybhna4-MuiTableCell-root"
              scope="col"
            >
              Score
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1imodrk-MuiTableCell-root"
              scope="col"
            >
              In stock
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-ybhna4-MuiTableCell-root"
              scope="col"
            >
              Price
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1g6gaq7-MuiTableCell-root"
              scope="col"
            >
              Visible
            </th>
          </tr>
        </thead>
        <tbody
          class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
          data-rbd-droppable-context-id="0"
          data-rbd-droppable-id="droppableTable"
        >
          <tr
            class="MuiTableRow-root css-9mllys-MuiTableRow-root"
            data-rbd-draggable-context-id="0"
            data-rbd-draggable-id="1"
          >
            <td
              aria-describedby="rbd-hidden-text-0-hidden-text-0"
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1sm79u7-MuiTableCell-root"
              data-rbd-drag-handle-context-id="0"
              data-rbd-drag-handle-draggable-id="1"
              draggable="false"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiBox-root css-siks6p"
              >
                <ion-icon
                  name="reorder-two-outline"
                />
              </div>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1h3ont0-MuiTableCell-root"
            >
              <div
                class="css-12hsf1n"
              >
                VA03
              </div>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1h3ont0-MuiTableCell-root"
            >
              <img
                alt="field.productImage"
                class="css-1qpslqa"
                src="static/media/assets/img/scarf_elastic.png"
              />
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1h3ont0-MuiTableCell-root"
            >
              <div
                class="css-12hsf1n"
              >
                Product name
              </div>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1h3ont0-MuiTableCell-root"
            >
              <div
                class="MuiBox-root css-0"
              >
                <div
                  class="MuiBox-root css-1sm7xfn"
                >
                  100.11
                </div>
              </div>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1h3ont0-MuiTableCell-root"
            >
              <div
                class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorSuccess MuiChip-filledSuccess css-s7mah8-MuiChip-root"
              >
                <span
                  class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                >
                  stock.inStock
                </span>
              </div>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1h3ont0-MuiTableCell-root"
            >
              <div
                class="MuiBox-root css-1u92jb9"
              >
                 
                10,00 €
              </div>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1h3ont0-MuiTableCell-root"
            >
              <span
                class="MuiSwitch-root MuiSwitch-sizeMedium css-tnsvld-MuiSwitch-root"
              >
                <span
                  aria-disabled="true"
                  class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-disabled Mui-disabled PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-disabled Mui-checked Mui-disabled css-1vk7jze-MuiButtonBase-root-MuiSwitch-switchBase"
                  tabindex="-1"
                >
                  <input
                    checked=""
                    class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
                    disabled=""
                    type="checkbox"
                  />
                  <span
                    class="MuiSwitch-thumb css-epwt2k-MuiSwitch-thumb"
                  />
                </span>
                <span
                  class="MuiSwitch-track css-1mr4grv-MuiSwitch-track"
                />
              </span>
            </td>
          </tr>
          <tr
            class="MuiTableRow-root css-9mllys-MuiTableRow-root"
            data-rbd-draggable-context-id="0"
            data-rbd-draggable-id="2"
          >
            <td
              aria-describedby="rbd-hidden-text-0-hidden-text-0"
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1sm79u7-MuiTableCell-root"
              data-rbd-drag-handle-context-id="0"
              data-rbd-drag-handle-draggable-id="2"
              draggable="false"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiBox-root css-siks6p"
              >
                <ion-icon
                  name="reorder-two-outline"
                />
              </div>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1h3ont0-MuiTableCell-root"
            >
              <div
                class="css-12hsf1n"
              >
                VA03
              </div>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1h3ont0-MuiTableCell-root"
            >
              <img
                alt="field.productImage"
                class="css-1qpslqa"
                src="static/media/assets/img/scarf_elastic.png"
              />
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1h3ont0-MuiTableCell-root"
            >
              <div
                class="css-12hsf1n"
              >
                Product name
              </div>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1h3ont0-MuiTableCell-root"
            >
              <div
                class="MuiBox-root css-0"
              >
                <div
                  class="MuiBox-root css-1sm7xfn"
                >
                  100.11
                </div>
              </div>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1h3ont0-MuiTableCell-root"
            >
              <div
                class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorError MuiChip-filledError css-1ywqr3b-MuiChip-root"
              >
                <span
                  class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                >
                  stock.outOfStock
                </span>
              </div>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1h3ont0-MuiTableCell-root"
            >
              <div
                class="MuiBox-root css-1u92jb9"
              >
                 
                10,00 €
              </div>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1h3ont0-MuiTableCell-root"
            >
              <span
                class="MuiSwitch-root MuiSwitch-sizeMedium css-tnsvld-MuiSwitch-root"
              >
                <span
                  aria-disabled="true"
                  class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-disabled Mui-disabled PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-disabled Mui-checked Mui-disabled css-1vk7jze-MuiButtonBase-root-MuiSwitch-switchBase"
                  tabindex="-1"
                >
                  <input
                    checked=""
                    class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
                    disabled=""
                    type="checkbox"
                  />
                  <span
                    class="MuiSwitch-thumb css-epwt2k-MuiSwitch-thumb"
                  />
                </span>
                <span
                  class="MuiSwitch-track css-1mr4grv-MuiSwitch-track"
                />
              </span>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
